lib/models/src/main/java/org/sagebionetworks/repo/model/dao/table/TableType.java

```diff
@@ -0,0 +1,67 @@
+package org.sagebionetworks.repo.model.dao.table;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.sagebionetworks.repo.model.EntityType;
+import org.sagebionetworks.repo.model.ObjectType;
+import org.sagebionetworks.repo.model.table.ViewEntityType;
+import org.sagebionetworks.util.ValidateArgument;
+
+/**
+ * Enumeration that maps all of the types of tables/views to {@link EntityType},
+ * {@link ObjectType}, and {@link ViewEntityType}
+ *
+ */
+public enum TableType {
+
+	table(ObjectType.TABLE),
+	entityview(ObjectType.ENTITY_VIEW),
+	submissionview(ObjectType.ENTITY_VIEW),
+	dataset(ObjectType.ENTITY_VIEW),
+	datasetcollection(ObjectType.ENTITY_VIEW),
+	materializedview(ObjectType.MATERIALIZED_VIEW);
+
+	private final ObjectType objectType;
+	private final EntityType entityType;
+	private final ViewEntityType viewEntityType;
```

Comment:
COnsider getting rid of the ViewEntityType, this enum could embed that information by itself